api/buildstatus/models/buildstatus.go

```diff
@@ -0,0 +1,118 @@
+package models
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strings"
+	"text/template"
+
+	v1 "github.com/equinor/radix-operator/pkg/apis/radix/v1"
+	"github.com/marstr/guid"
+)
+
+const BUILD_STATUS_SVG_RELATIVE_PATH = "./badges/build-status.svg"
+const BUILD_STATUS_FAILING = "failing"
+const BUILD_STATUS_PASSING = "passing"
+const BUILD_STATUS_STOPPED = "stopped"
+const BUILD_STATUS_PENDING = "pending"
+const BUILD_STATUS_UNKNOWN = "unknown"
+
+type Status interface {
+	WriteSvg(condition v1.RadixJobCondition) (*[]byte, error)
+}
+
+func NewBuildStatus() Status {
+	return &radixBuildStatus{
+		Operation:   "build",
+		ColorLeft:   "#aaa",
+		ColorShadow: "#010101",
+		ColorFont:   "#fff",
+	}
+}
+
+type radixBuildStatus struct {
+	Operation       string
+	Status          string
+	ColorLeft       string
+	ColorRight      string
+	ColorShadow     string
+	ColorFont       string
+	Width           int
+	Height          int
+	StatusOffset    int
+	OperationTextId string
+	StatusTextId    string
+}
+
+func (rbs *radixBuildStatus) WriteSvg(condition v1.RadixJobCondition) (*[]byte, error) {
+	rbs.Status = translateCondition(condition)
+	color := getColor(rbs.Status)
+	rbs.ColorRight = color
+	return getStatus(rbs)
+}
+
+func translateCondition(condition v1.RadixJobCondition) string {
+	if condition == v1.JobSucceeded {
+		return BUILD_STATUS_PASSING
+	} else if condition == v1.JobFailed {
+		return BUILD_STATUS_FAILING
+	} else if condition == v1.JobStopped {
+		return BUILD_STATUS_STOPPED
+	} else if condition == v1.JobWaiting || condition == v1.JobQueued {
+		return BUILD_STATUS_PENDING
+	} else {
+		return BUILD_STATUS_UNKNOWN
+	}
+}
+
+func getStatus(status *radixBuildStatus) (*[]byte, error) {
+	operationWidth := calculateWidth(9, status.Operation)
+	statusWidth := calculateWidth(12, status.Status)
+	status.Width = statusWidth + operationWidth
+	status.Height = 30
+	status.StatusOffset = operationWidth
+	status.OperationTextId = guid.NewGUID().String()
+	status.StatusTextId = guid.NewGUID().String()
+	svgTemplate, err := template.ParseFiles(BUILD_STATUS_SVG_RELATIVE_PATH)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
log.Fatal will cause application to panic and exit. Change to log.Error and return the err to caller.